test/test_routes.py

```diff
@@ -586,6 +586,31 @@ def identity(name, request: gr.Request):
         output = dict(response.json())
         assert output["data"] == ["test"]
 
+    def test_request_argument_passing(self):
+        def identity(message, req: gr.Request, history):
+            return message
```

Comment:
I think this method needs to include something like this:

https://github.com/gradio-app/gradio/blob/afcf3c48e82712067d6d00a0caedb1562eb986f8/test/test_routes.py#L521

to make sure req is actually passed in